### PR TITLE
Increase User Friendliness

### PR DESCRIPTION
**Pull request by t0mdavid-m**
This PR aims to increase usability by adding the following features:

- [x] Separate out workflows into multiple pages
- [x] Move sharing of workspace id to address bar as get parameter
	- Add `get` parameter
	- Remove workspace selection from sidebar in hosted mode
- [x] Add google analytics for usage tracking
	- Add custom component using `Klaro` as consent manager
	- Add `iframe` providing tracking data
- [x] Add a `.yml`/`.json` configuration file
	- Opted for `json` as `yml` would have required the addition of a new dependency
	- Allow configuration of Google Analytics Tracking ID and enabling/disabling of google analytics
	- control of online/offline deployment
- [x] Update of documentation
- [x] Separate common modules from the template source files intended to be modified by developers

While I was at it I also fixed the following issues:
- Streamlit execution was broken:  
	- Removed `st.fragment` decorator from execution section
- Streamlit logging was broken: 
	- Reintroduced `st.rerun`with a three second time delay after execution
	- Protected `app.py` for multiprocessing
- `st.experimental_fragment` is now `st.fragment`
- Fixed failing CI pipelines
	- Fixed test dependency on old Streamlit version
	- Fixed artifact versioning for windows build
- The PyOpenMS workflow threw an exception if the results directory had not been created yet.
- The filepaths in the download section were broken

### CODE REVIEW

File: content/quickstart.py [hunk 1]
```diff
@@ -77,11 +77,11 @@
 
 st.markdown(
     """##  Workspaces and Settings
-The **sidebar** contains to boxes, one for **workspaces** and one for **settings**.
+The **sidebar** contains to boxes, one for **workspaces** (in local mode) and one for **settings**.
 
 🖥️ **Workspaces** store user inputs, parameters and results for a specific session or analysis task.
 
-In **online mode** where the app is hosted on a remote server the workspace has a unique identifier number which can be shared with collaboration partners or stored for later access.
+In **online mode** where the app is hosted on a remote server the workspace has a unique identifier number which can be shared with collaboration partners or stored for later access. The identifier is embedded within the url.
```
Comment (axelwalter) [inline content/quickstart.py:84]:
```suggestion
In **online mode** where the app is hosted on a remote server the workspace has a unique identifier number embedded within the URL. To share your data analysis with collaboration partners simply share the URL.
```

File: docs/toppframework.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def content():
 
 The subdirectory name will be determined by a **key** that is defined in the `self.ui.upload_widget` method. The uploaded files are available by the specific key for parameter input widgets and accessible while building the workflow.
 
-Calling this method will create a complete file upload widget section with the following components:
+Calling this method will create a complete file upload widget page with the following components:
```
Comment (axelwalter) [inline docs/toppframework.py:64]:
```suggestion
Calling this method will create a complete file upload page with the following components:
```

**Review by axelwalter — APPROVED** [review]
Thanks for the great work with useful updates and additions!

**Comment by timosachsenberg** [comment]
can we somehow add a test that can verify that an app can be executed? e.g., is there a simple testing framework that allows to "click on load example data" , then does a bit of processing and clicking through menues?

**Comment by axelwalter** [comment]
> can we somehow add a test that can verify that an app can be executed? e.g., is there a simple testing framework that allows to "click on load example data" , then does a bit of processing and clicking through menues?

The integrated Streamlit App testing should cover a lot:
https://docs.streamlit.io/develop/api-reference/app-testing

File: src/common/__pycache__/captcha_.cpython-311.pyc
[the remark this answers is not in the document]
Comment (t0mdavid-m) [reply src/common/__pycache__/captcha_.cpython-311.pyc:1]: Done!

**Comment by t0mdavid-m** [comment]
Thank you for the suggestions @timosachsenberg @axelwalter . I will look into including (a) simple test(s)!

**Pull request merged**